src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -175,59 +218,89 @@ public Task<ActionResult<IEnumerable<ProcessIdentifierModel>>> GetProcesses()
+            int metricsIntervalSeconds = 1,
+            [FromQuery]
+            string egressProvider = null)
         {
             return InvokeForProcess(processInfo =>
             {
                 TimeSpan duration = ConvertSecondsToTimeSpan(durationSeconds);
 
                 var configurations = new List<MonitoringSourceConfiguration>();
-                if (profile.HasFlag(TraceProfile.Cpu))
+                if (profile.HasFlag(Models.TraceProfile.Cpu))
                 {
                     configurations.Add(new CpuProfileConfiguration());
                 }
-                if (profile.HasFlag(TraceProfile.Http))
+                if (profile.HasFlag(Models.TraceProfile.Http))
                 {
                     configurations.Add(new HttpRequestSourceConfiguration());
                 }
-                if (profile.HasFlag(TraceProfile.Logs))
+                if (profile.HasFlag(Models.TraceProfile.Logs))
                 {
                     configurations.Add(new LoggingSourceConfiguration());
                 }
-                if (profile.HasFlag(TraceProfile.Metrics))
+                if (profile.HasFlag(Models.TraceProfile.Metrics))
                 {
                     configurations.Add(new MetricSourceConfiguration(metricsIntervalSeconds, Enumerable.Empty<string>()));
                 }
 
                 var aggregateConfiguration = new AggregateSourceConfiguration(configurations.ToArray());
 
                 return StartTrace(processInfo, aggregateConfiguration, duration, egressProvider);
-            }, processFilter, ArtifactType_Trace);
+            }, processKey, ArtifactType_Trace);
         }
 
-        [HttpPost("trace/{processFilter?}")]
-        public Task<ActionResult> TraceCustomConfiguration(
-            ProcessFilter? processFilter,
-            [FromBody][Required] EventPipeConfigurationModel configuration,
-            [FromQuery][Range(-1, int.MaxValue)] int durationSeconds = 30,
-            [FromQuery] string egressProvider = null)
+        /// <summary>
+        /// Capture a trace of a process.
+        /// </summary>
+        /// <param name="processKey">Value used to identify the target process, either the process ID or the runtime instance cookie.</param>
+        /// <param name="configuration">The trace configuration describing which events to capture.</param>
+        /// <param name="durationSeconds">The duration of the trace session (in seconds).</param>
+        /// <param name="egressProvider">The egress provider to which the trace is saved.</param>
+        [HttpPost("trace/{processKey?}", Name = nameof(CaptureTraceCustom))]
+        [Produces(ContentTypes.ApplicationOctectStream, ContentTypes.ApplicationProblemJson)]
+        // FileResult is the closest representation of the output so that the OpenAPI document correctly
+        // describes the result as a binary file.
+        [ProducesResponseType(typeof(FileResult), StatusCodes.Status200OK)]
```

Comment:
IIRC part of the Api review was to produce 202's if we egress. That doesn't have to be done right now, but I'm assuming that's something we can easily add later.

Reply:
Correct, that should be "easily" done, but I do not want to do that at this point. I want to wait until we get agreement on the HTTP surface area before making any significant change.